pkg/alertmanager/alertmanager.go

```diff
@@ -637,7 +637,8 @@ func (am *Alertmanager) buildIntegrationsMap(emailCfg alertingReceivers.EmailSen
 	for _, rcv := range nc {
 		var integrations []*nfstatus.Integration
 		var err error
-		if rcv.Type() == definition.GrafanaReceiverType {
+		rcvType := rcv.Type()
```

Comment:
Wdyt about using a `switch` statement with cases for Grafana and Mimir receivers, adding a `default` statement with a comment that makes it clear that we're not building a `tmpl` object for empty receivers?

```go
switch rcv.Type() {
    case GrafanaReceiverType:
        // Logic here...
    case AlertmanagerReceiverType:
        // Logic here...
    default:
        // Comment explaining that we're not creating a tmpl for this case.
}
```